### PR TITLE
170947604-Allow user to reset password

### PR DESCRIPTION
**Pull request by EugeneMunya**
#### What does this PR do?
- presenting a user the first form to reset password
#### Description of Task to be completed?
- a form that allows user to enter his/her email in order to reset password
#### How should this be manually tested?
- you can you this link `https://knights-bn-f-ft-reset-p-zwf9bi.herokuapp.com/` to testing it online or follow the steps below to test it in your local computer
- opening your terminal and run this command `git clone https://github.com/Stackup-Rwanda/knights-bn-frontend.git` 
- run `cd knights-bn-frontend folder`
 - run `git fetch && git checkout ft-reset-password-170947604` 
- run `npm install` and then run `npm run dev-start`
- go to login
- Click on forgot password link
- enter your email to request a password modification
- after you successfully request to reset the password you will receive an email that contains a link you will use to write new credentials, then click on that link and it will take you to the new browser tab. since we are testing locally change the URL where there is `https://knights-bn-frontend.herokuapp.com` replace it with `http://localhost:8080` and type your new credentials and send.
#### What are the relevant pivotal tracker stories?
[#170947604](https://www.pivotaltracker.com/story/show/170947604)
#### Screenshots (if appropriate)
Request to reset password

Screen Shot 2020-05-06 at 09 07 58

### CODE REVIEW

**Comment by Niyonsengaeric** [comment]
1. Squash your commit
2. screenshots are not matching with what I get to my end try to change it
3. try to redirect either to home or login  after success reset

**Comment by EugeneMunya** [comment]
> 1. Squash your commit
> 2. screenshots are not matching with what I get to my end try to change it
> 3. try to redirect either to home or login  after success reset

I have done it you can check again

**Review by T2Wil — APPROVED** [review]
Nice work @EugeneMunya .

**Review by Niyonsengaeric — APPROVED** [review]
@EugeneMunya  Good work!

**Review by Kagorora — APPROVED** [review]

File: __tests__/components/resetPassword/resetPasswordform.test.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import { act } from 'react-dom/test-utils';
+import ResetPaswordForm from '../../../src/components/resetPassword/ResetPasswordForm';
+
+describe('Test reset password', () => {
+  it('form test', async () => {
+    const displayMessage = '';
+    const checkOnsubmit = jest.fn();
+    const callApiThunk = jest.fn();
+    const location = jest.fn();
+    const checkOnchange = jest.fn();
```
Comment (kagabof) [inline __tests__/components/resetPassword/resetPasswordform.test.js]: Please avoid declaring a lot of variables. You can use objects.

**Review by kagabof — CHANGES_REQUESTED** [review]
Please add your deployment link.
check your tests they are throwing some errors.

File: __tests__/components/resetPassword/resetPasswordform.test.js (hunk 1 above)
Comment (EugeneMunya) [reply __tests__/components/resetPassword/resetPasswordform.test.js]: I let me do it now

**Comment by EugeneMunya** [comment]
> Please add your deployment link.
> check your tests they are throwing some errors.

The errors are not min is someone who nested the form

File: __tests__/components/resetPassword/resetPasswordform.test.js (hunk 1 above)
Comment (EugeneMunya) [reply __tests__/components/resetPassword/resetPasswordform.test.js]: The errors were not mine but we will refactor the after

**Review by kagabof — APPROVED** [review]
Well done.

File: __tests__/components/resetPassword/resetPasswordform.test.js (hunk 1 above)
Comment (kagabof) [reply __tests__/components/resetPassword/resetPasswordform.test.js]: cool.

**Pull request merged**